scripts/consolidate-memory.py

```
@@ -0,0 +1,492 @@
+Issues: #153
+"""
+
+import argparse
+import json
+import re
+from collections import Counter
+from dataclasses import dataclass, field
+from datetime import date, datetime, timedelta, timezone
+from math import log
+from pathlib import Path
+from typing import Set
```

Ruff (0.14.14)

[warning] `typing.Set` is deprecated, use `set` instead

(UP035)